src/Microsoft.Data.SqlClient/netfx/src/Microsoft/Data/SqlClient/TdsParserStateObject.netfx.cs

```diff
@@ -887,12 +886,6 @@ internal bool IsConnectionAlive(bool throwOnException)
                     }
                     finally
                     {
-                        if (readPacket != IntPtr.Zero)
-                        {
-                            // Be sure to release packet, otherwise it will be leaked by native.
-                            SNINativeMethodWrapper.SNIPacketRelease(readPacket);
```

Comment:
Since this code is unreachable, try/finally serves no purpose other than whatever RuntimeHelpers.PrepareConstrainedRegions does. I have preserved the try/finally (by adding it in .net core) but I suspect it can be removed, because according to the [documentation](https://learn.microsoft.com/en-us/dotnet/framework/performance/constrained-execution-regions) (emphasis mine):

> The [PrepareConstrainedRegions](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.compilerservices.runtimehelpers.prepareconstrainedregions) method must immediately precede a try block and **_marks catch, finally, and fault blocks_** as constrained execution regions.